src/cpp/core/r_util/RProjectFile.cpp

```diff
@@ -311,6 +312,54 @@ std::ostream& operator << (std::ostream& stream, const YesNoAskValue& val)
    return stream ;
 }
 
+Error findProjectFile(FilePath filePath,
+                      FilePath* pProjPath)
+{
+   // check to see if we already have a .Rproj file
+   if (filePath.extension() == ".Rproj")
+   {
+      *pProjPath = filePath;
+      return Success();
+   }
+   
+   if (!filePath.isDirectory())
+      filePath = filePath.parent();
+   
+   // no .Rproj file found; scan parent directories
+   for (; filePath.exists(); filePath = filePath.parent())
```

Comment:
I think it would be simpler just to provide a 'quiet' API that doesn't log errors.